sa/sa.go

```diff
@@ -947,17 +942,39 @@ func (ssa *SQLStorageAuthority) CountCertificatesRange(ctx context.Context, star
 }
 
 // CountPendingAuthorizations returns the number of pending, unexpired
-// authorizations for the give registration.
-func (ssa *SQLStorageAuthority) CountPendingAuthorizations(ctx context.Context, regID int64) (count int, err error) {
-	err = ssa.dbMap.SelectOne(&count,
-		`SELECT count(1) FROM pendingAuthorizations
-		 WHERE registrationID = :regID AND
-				expires > :now`,
-		map[string]interface{}{
-			"regID": regID,
-			"now":   ssa.clk.Now(),
-		})
-	return
+// authorizations for the given registration.
+func (ssa *SQLStorageAuthority) CountPendingAuthorizations(ctx context.Context, regID int64) (int, error) {
+	var count int64
+
+	/*
+	 * We need to look at *both* the `authz` table and the `pendingAuthorizations`
+	 * table during the transition period described in Issue 2162[0]
+	 *
+	 * [0] - https://github.com/letsencrypt/boulder/issues/2162
+	 */
+	for _, table := range authorizationTables {
+		stmtArgs := []interface{}{regID, ssa.clk.Now()}
+		qmarks := []string{}
+		for _, status := range pendingStatuses {
```

Comment:
Since pendingStatuses is fairly static, I would be happier hardcoding `"?,?,?"` and the list of pending statuses (via their const names). That would simplify this code and require less reasoning about its correctness.

If you want to defend against future addition of a status that is considered pending, you can add a unittest that checks that pendingStatuses = the current list, with an error message that CountPendingAuthorizations needs to be changed if that changes.